### ISSUE TITLE
Workspaces, applications, laucher icons do not open on touch screen with finger select. 

### ISSUE THREAD

**Issue by math93033** [pop-os/cosmic-dock#181]

**Comment by jacobgkau** [comment]
Not sure this is cosmic-dock if it's also the top bar, could be the main COSMIC extension.

**Comment by jacobgkau** [comment]
@math93033 I assume from your description that other apps open from the dock normally?

**Comment by leviport** [comment]
I tested on a different model of laptop with a touch screen. I can kinda get them to work, but they're definitely inconsistent. Applications in particular seems iffy. I see it pop up briefly, then go away. I'm not sure if it's just me, but it seems like they work better on an empty workspace, with no windows already open. Any app that also summons the on-screen keyboard seems to make the behavior even more strange.

It seems like the Workspaces and Applications buttons on the top bar are a lot more reliable, but the App Library does sometimes disappear on me. It could definitely be the main extension though.

**Comment by math93033** [comment]
@leviport  Yes. Everything else opens correctly. It's only the pop specific dock and top bar navigation icons that do not work.
